lib/setup/setupInstall.js

```diff
@@ -1545,28 +1550,35 @@ function Install(options) {
                 const githubParts = url.match(/^https:\/\/github\.com\/([^/]+\/[^/]+)\/?.*$/);
                 if (githubParts && githubParts[1]) {
                     request.get({
-                        url: 'http://api.github.com/repos/' + githubParts[1] + '/commits/master',
+                        url: `http://api.github.com/repos/${githubParts[1]}/commits`,
                         json: true,
                         headers: {'User-Agent': 'ioBroker Adapter install'}
                     }, (err, res, data) => {
                         if (err) {
-                            console.log('Info: Can not get current GitHub commit, we remember master branch only: ' + err);
-                            return void that.installAdapterFromUrl(url + '/tarball/master', name, callback);
+                            console.log(`Info: Can not get current GitHub commit, only remember that we installed from GitHub: ${err}`);
+                            // Install using the npm Github URL syntax `npm i user/repo_name`:
+                            return void that.installAdapterFromUrl(githubParts[1], name, callback);
                         } else if (res.statusCode !== 200) {
-                            console.log('Info: Can not get current GitHub commit, we remember master branch only. Status:', res.statusCode + (data && data.message ? ' (' + data.message + ')' : ''));
-                            return void that.installAdapterFromUrl(url + '/tarball/master', name, callback);
-                        } else if (!data.sha) {
-                            console.log('Info: Can not get current GitHub commit, we remember master branch only. No SHA available');
-                            return void that.installAdapterFromUrl(url + '/tarball/master', name, callback);
+                            console.log(`Info: Can not get current GitHub commit, only remember that we installed from GitHub. Status: ${res.statusCode}${data && data.message ? ` (${data.message})` : ''}`);
+                            // Install using the npm Github URL syntax `npm i user/repo_name`:
+                            return void that.installAdapterFromUrl(githubParts[1], name, callback);
+                        } else if (data && Array.isArray(data) && data.length >= 1 && data[0].sha) {
+                            // Install using the npm Github URL syntax `npm i user/repo_name#commit-ish`:
+                            return void that.installAdapterFromUrl(`${githubParts[1]}#${data[0].sha}`, name, callback);
```

Comment:
Sorry, this will not work that way ... we need an URL ... "iobrober install" command only supports @ to provide a version; and "iobroker url" expects an URL.

Alternatively we need to enhance ibroker install to also allow the github style ... maybe then with github:user/repo#hash-or-branch

Reply:
The code that uses the installedFrom is: https://github.com/ioBroker/ioBroker.js-controller/blob/master/main.js#L2861-L2871 and till "Only" calls iobroker url/install commands

Reply:
The CLI command `iobroker url` just checks if an argument was given and if yes, calls `installAdapterFromUrl`, which leads here. We should fix the other locations to accept `user/repo#commit-ish` URLs aswell then. I'm against adding a custom prefix, because then `npm` doesn't understand it and we need to strip it again (potentially causing us to forget it).

Reply:
And what's the problem with `iobroker install`? I thought that is supposed to take the adapter name, not an URL.